### PR TITLE
document create/grant/revoke/delete and other cleanup; closes #206; closes #209; closes #211

### LINKED ISSUE #206: [docs] create/grant/revoke/delete

### LINKED ISSUE #209: [docs] anatomy of a request
from client (cli/slack/etc...) -> server, check config, determine identity, execute action on target, return response. talk about the config check and identity determination.

### LINKED ISSUE #211: [docs] cleanup
- Document PO_TOKEN
- Remove "admin" from "please admin encrypt" https://docs.pleaseops.com/configuration/credentials/

### PR DESCRIPTION
**Pull request by jonnymbgood**

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -44,10 +44,18 @@ Permissions match the Action/Target scheme, where each user must be explicitly g
...
 Multiple clients (Slack, CLI, etc...) connect to a running PleaseOps Server, which is connected to a configuration repository (git). It is possible to run PleaseOps in HA mode by running 2+ servers behind a load balancer.
 
-
 ### Configuration
 Configuration is stored in a git repository and referenced by the server to determine available actions and targets. Secrets stored within the git repository are asymmetrically encrypted using Public/Private Key Cryptography such that only the server can read them. Any commits to the configuration repository are automatically picked up by the server before executing any command, ensuring it has the latest configuration and proper permissioning. For more details on configuration please see the documentation [here](./configuration/overview.md).
 
+### Anatomy of a Request
+1. A request is issued by a client (slack, cli, etc...) to the server.
+1. The server receives the request and ensures it has the latest config. (This config instance is used for the duration of the request).
+1. The server verified the validity of the request and identifies the user.
```
Comment (thuey) [inline docs/README.md]: *verifies

File: docs/README.md [hunk 2]
```diff
@@ -44,10 +44,18 @@ Permissions match the Action/Target scheme, where each user must be explicitly g
...
+      * Slack - the verification token is checked and the username is matched to the user's id.
+      * Cli - the jwt is verified and the name pulled from the `sub` jwt value and matched to the user's id.
```
Comment (thuey) [inline docs/README.md]: *is matched

File: docs/actions/create.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
+title: create
+---
+# create [type] [id]
+Create roles and users
+
+!!! important
+    This command will only work if you granted your deploy key write permissions.
```
Comment (thuey) [inline docs/actions/create.md]: Is "deploy key" what we are calling the jwt? Can you run this from slack?

File: docs/actions/create.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+## create role
+This command creates a role. This will fail if the role exists or contains a `.`.
+
+### Usage
+```
+$> please create role developer
+Created role foo
```
Comment (thuey) [inline docs/actions/create.md]: foo -> developer

File: docs/actions/delete.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+!!! important
+    This command will only work if you granted your deploy key write permissions.
+
+## delete role
+This command deletes a role. This will fail if the role does not exist.
+
+### Usage
+```
+$> please delete role developer
+Deleted role foo
```
Comment (thuey) [inline docs/actions/delete.md]: foo -> developer

File: docs/actions/grant.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+---
+title: grant
+---
+# grant [type] [id] [type] [id]
+Grant user's roles and role's permissions
```
Comment (thuey) [inline docs/actions/grant.md]: No '. Also, are we not supporting granting permissions to users directly?

File: docs/clients/cli.md [hunk 7]
```diff
@@ -5,7 +5,7 @@ title: CLI
 The cli is available for windows, mac, and linux. You can download it by visiting `<your.pleaseops.server>/v1/cli` and placing the binary in your `PATH`.
 
 ## Configuration
-By default the cli reads its config file from `$HOME/.pleaseops/config.yaml`. It also accepts `.json` and `.toml` formats. To use a different config file export the environment variable `PO_CONFIG=path/to/file`. You can set any of the configuration options by exporting environment variables prefixed with `PO_`. For example, `server` becomes `PO_SERVER`.
+By default the cli reads its config file from `$HOME/.pleaseops/config.yaml`. It also accepts `.json` and `.toml` formats. To use a different config file export the environment variable `PO_CONFIG=path/to/file`. You can set any of the configuration options by exporting environment variables prefixed with `PO_`. For example, `server` becomes `PO_SERVER` and `token` becomes `PO_TOKEN`.
```
Comment (thuey) [inline docs/clients/cli.md]: Should we document what the config options are? In particular, I don't think we explain anywhere what the token is or how to configure it

File: docs/README.md (hunk 1 above)
Comment (jonnymbgood) [reply docs/README.md]: done

File: docs/README.md (hunk 2 above)
Comment (jonnymbgood) [reply docs/README.md]: done

File: docs/actions/create.md (hunk 4 above)
Comment (jonnymbgood) [reply docs/actions/create.md]: done

File: docs/actions/delete.md (hunk 5 above)
Comment (jonnymbgood) [reply docs/actions/delete.md]: done

File: docs/clients/cli.md (hunk 7 above)
Comment (jonnymbgood) [reply docs/clients/cli.md]: They are documented below in the config section.

File: docs/actions/grant.md (hunk 6 above)
Comment (jonnymbgood) [reply docs/actions/grant.md]: Fixed. And not yet.

File: docs/actions/create.md (hunk 3 above)
Comment (jonnymbgood) [reply docs/actions/create.md]: fixed.

**Pull request merged**